File: backend/internal/domain/entities/trade.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Trade struct {
	ID             uuid.UUID  `json:"id"`
	UserID         uuid.UUID  `json:"user_id"`
	BubbleID       *uuid.UUID `json:"bubble_id,omitempty"`
	Exchange       string     `json:"exchange"`
	BinanceTradeID int64      `json:"binance_trade_id"`
	Symbol         string     `json:"symbol"`
	Side           string     `json:"side"`
	PositionSide   *string    `json:"position_side,omitempty"`
	OpenClose      *string    `json:"open_close,omitempty"`
	ReduceOnly     *bool      `json:"reduce_only,omitempty"`
	Quantity       string     `json:"quantity"`
	Price          string     `json:"price"`
	RealizedPnL    *string    `json:"realized_pnl,omitempty"`
	TradeTime      time.Time  `json:"trade_time"`
}

File: frontend/src/types/trade.ts
export type TradeItem = {
  id: string
  bubble_id?: string
  exchange: string
  symbol: string
  side: string
  position_side?: string
  open_close?: string
  reduce_only?: boolean
  quantity: string
  price: string
  realized_pnl?: string
  trade_time: string
  binance_trade_id: number
}

export type TradeListResponse = {
  page: number
  limit: number
  total: number
  items: TradeItem[]
}

export type TradeSideSummary = {
  side: string
  trade_count?: number
  total_trades?: number
  total_volume?: string
  realized_pnl_total?: string
}

export type TradeSymbolSummary = {
  symbol: string
  trade_count?: number
  total_trades?: number
  buy_count?: number
  sell_count?: number
  total_volume?: string
  realized_pnl_total?: string
  wins?: number
  losses?: number
}

export type TradeExchangeSummary = {
  exchange: string
  trade_count?: number
  total_trades?: number
  buy_count?: number
  sell_count?: number
  total_volume?: string
  realized_pnl_total?: string
}

export type TradeTotals = {
  total_trades: number
  buy_count?: number
  sell_count?: number
  total_volume?: string
  realized_pnl_total?: string
  wins?: number
  losses?: number
  breakeven?: number
  average_pnl?: string
}

export type TradeSummaryResponse = {
  exchange: string
  totals: TradeTotals
  by_exchange: TradeExchangeSummary[]
  by_side: TradeSideSummary[]
  by_symbol: TradeSymbolSummary[]
}
